pkg/bgpv1/test/adverts_test.go

```diff
@@ -0,0 +1,360 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	slim_metav1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	"github.com/cilium/cilium/pkg/testutils"
+
+	"github.com/stretchr/testify/require"
```

Comment:
So motivation is two fold, and goes hand in hand with current unit tests.
- Increase test coverage around boundaries of subcomponents
    - For eg, we do not test complete flow of change from getting Kubernetes event and then validating that it is reflected in the peering router.
    - Unit tests today exists for 
        - Controller : This is limited to that policy change calls correct reconciler methods.
        - Reconcilers : Tests validate the diffs are generated correctly and we are inspecting gobgp directly for validation. 
            - Direct inspection of gobgp might not be suitable here in long term, if another underlying routing agent is used.
- Validate and make BGP Peering part of CI testing.
    - This is mainly for neighbour configuration : like timers, graceful restart, authentication etc. And later on for more advanced BGP as well ( Routing policies and filters )
    - During new feature development, we have to validate the expected behaviour and add tests for it. It is currently not possible with unit tests, as we only configure local gobgp and we have to validate the behaviour separately via container labs - to test actual peering behaviour. 
    - Having this tests as part of CI, will ensure that there are no regressions introduced as we make changes.

Reply:
Ah, I was just asking the reason for introducing `github.com/stretchr/testify/require` for curiosity 😓 

But, thanks for the clarification. I think it would be nice to have these contexts in the commit message.

Reply:
Ah, so there are two packages in testify, assert and require. From what I understand, on any failure, assert will continue the test run and execute as many sub-tests as possible and eventually fail the test case. 

Whereas, require will fail the test case on failing step and not proceed further. I personally prefer this approach, so time and resources are not spent on test case which is failing anyways. Also, it provides quicker feedback while doing development.

Reply:
Also, testify package provides interesting validation patterns. Like comparing sets, values, error/noerror etc. Which makes test more readable.

Reply:
Alright, that makes sense. I don't really have any preference around the testing framework, so no objection 👍